onnxscript/rewriter/onnxruntime/bfloat16_utils/bfloat16_converter.py

```diff
@@ -0,0 +1,92 @@
+import logging
+
+from onnxscript import ir
+
+logger = logging.getLogger(__name__)
+CREATED_CAST_BFLOAT16_NAME_SUFFIX = "_cast_bfloat16"
+
+
+def _convert_inputs_from_bfloat16_to_float16(value: ir.Input) -> None:
+    if value.dtype != ir.DataType.BFLOAT16:
+        return
+    value.dtype = ir.DataType.FLOAT16
+    value_users = tuple(value.uses())
+    for node, index in value_users:
+        _insert_cast_nodes_for_float16_to_bfloat16_to_inputs(node, index)
```

Comment:
If the var is used in 100 nodes, we don't want to create 100 cast nodes? We should create 1 cast node, and replace all uses of original_var by cast_var.

Reply:
Make sense